### PR TITLE
Chana/migrate loader to ts

### PR DESCRIPTION
**Pull request by juanigaray**

### CODE REVIEW

**Comment by xaiki** [comment]
probably merge this with #92 ?

**Comment by juanigaray** [comment]
> probably merge this with #92 ?

I actually created the issue because of this PR and #90 ! It'd have to be solved in a different PR, in my opinion, as it falls outside of the scope of a) migrating the codebase to TS and b) fixing the cascade

**Comment by xaiki** [comment]
changing the place where things are defined is not strictly migrating to TS, and this particular PR doesn't help with the cascade.

**Comment by juanigaray** [comment]
> changing the place where things are defined is not strictly migrating to TS, and this particular PR doesn't help with the cascade.

Converting the unused Loader component would've been useless, and I felt that leaving it as JS would've been insufficient, so I deleted it, from which the only logical conclusion was to move this module and type it.


Do you believe this to be off-scope from `feature/typescript`? What is your proposal?

**Comment by xaiki** [comment]
we agreed to merge this with #90 and fork `feature/performance` from it.

File: src/helpers/fetchUrlsData.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   * on one hand, the key of the URL,
+   * and on the other, the response to the URL itself
+   */
+  const keyAndResponsePromises = Object.entries(urls).map(
+    async ([key, url]): Promise<[string, string]> => [
+      key,
+      await fetchSingleUrlData(url),
+    ],
+  );
+  const keyAndResponses = await Promise.all(keyAndResponsePromises);
```
Comment (xaiki) [inline src/helpers/fetchUrlsData.ts:18]: this leaves #92 unresolved

Comment (xaiki) [reply src/helpers/fetchUrlsData.ts:18]: either put a comment saying that we know there is a bug
or use allSettled

Comment (juanigaray) [reply src/helpers/fetchUrlsData.ts:18]: There isn't a bug in my opinion, there's a potential improvement which is already tracked in an issue.

I'd add a TODO linking the issue but you specifically asked that we don't do things like this. "That's what issues are for. This adds clutter to the PR and to the code. PRs should do one thing only. Leaving comments for the future is not cool".

I'm not sure what to do if we want to be consistent.

Comment (xaiki) [reply src/helpers/fetchUrlsData.ts:18]: a failing await will throw, and that is picked up by react router
changing it to Promise.all() makes it NOT THROUGH, and hence fail silently.

**Review by xaiki — APPROVED** [review]
Promise.all() does reject, so we have no API change.
my bad.

lgtm

**Pull request merged**